### PR TITLE
Revert "Revert Factory Registration to fix ListView regression (#17084)"

### PR DESCRIPTION
**Pull request by jonathanpeppers**
This reverts commit 6bc5ea911dcd4dfda99f446d8ac01fc00ae35a25.

Will manually run the UI Tests here to understand what was going wrong with the original changes.

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
UI Tests (if I did it right): https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8367136&view=results

**Comment by rmarinho** [comment]
image

Yeah those Android ones are the regressions @jonathanpeppers  , Can you try run the controll gallery locally with this change?

File: src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs [hunk 1]
```diff
@@ -62,101 +62,118 @@ public static partial class AppHostBuilderExtensions
...
+			handlersCollection.AddHandler<ActivityIndicator>(_ => new ActivityIndicatorHandler());
+			handlersCollection.AddHandler<BoxView>(_ => new BoxViewHandler());
+			handlersCollection.AddHandler<Button>(_ => new ButtonHandler());
+			handlersCollection.AddHandler<CheckBox>(_ => new CheckBoxHandler());
+			handlersCollection.AddHandler<DatePicker>(_ => new DatePickerHandler());
+			handlersCollection.AddHandler<Editor>(_ => new EditorHandler());
+			handlersCollection.AddHandler<Entry>(_ => new EntryHandler());
+			handlersCollection.AddHandler<GraphicsView>(_ => new GraphicsViewHandler());
+			handlersCollection.AddHandler<Image>(_ => new ImageHandler());
+			handlersCollection.AddHandler<Label>(_ => new LabelHandler());
```
Comment (jonathanpeppers) [inline src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs:77]: If I put this line back, it fixes Tim's app: https://github.com/dotnet/maui/issues/17028

Makes no sense, feels like we are using Microsoft.Extensions.DI wrong?

**Comment by jonathanpeppers** [comment]
This isn't going to be able to work, the problem is this line:

https://github.com/dotnet/maui/blob/65edfe095cc3fed824b83ad76bed755b17bd52a1/src/Core/src/Hosting/Internal/MauiHandlersFactory.cs#L42

`ImplementationType` is null because `ImplementationFactory` should be used instead.

I reworked it to use:

https://github.com/dotnet/maui/blob/1f3302945f8049a63f5c3345909756c43651e247/src/Core/src/Hosting/Internal/MauiHandlersFactory.cs#L56-L74

But this won't work either because the return type is `IElementHandler` and this code needs the _concrete_ type to work.

I don't think this change can work, even if I do:

```csharp
handlersCollection.AddTransient(typeof(Label), _ => new LabelHandler());
```
In this case the return type is `object`...

In the future, it feels like Microsoft.Extensions.DependencyInjection usage in MAUI should be generally refactored. Too much code is using `ServiceDescriptor` directly. If we want to use Microsoft.Extensions.DI, we should just use the DI container normally like customers would.

I pushed my change just to see what happens on CI, but Tim's app still has an issue.

**Comment by jonathanpeppers** [comment]
New test run: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8371885&view=results

**Pull request closed without merging**